app/controllers/users_controller.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])  #form tags
+    @user = User.new(user_params) # form_for
+    if @user.save
+      redirect_to new_user_path
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
```

Comment:
- [ ] What if something goes wrong on the update action? We can do as we did in the create action to check if the user is saved successfully.